EC-84: Revise Configuration, Customers, eContent, Login and Logging (#5)
ezsystems/commerce-documentation: docs/guide/econtent/econtent_faq.md

# eContent FAQ

## After using eContent the top navigation is empty

Check your data inside eContent.

> Make sure that for all objects the attribute which is defined as `name_identifier` in `sve_class` is stored in `sve_object_attributes`

Reviewer: ```suggestion
Make sure that for all objects the attribute that is defined as `name_identifier` in `sve_class` is stored in `sve_object_attributes`
```

+---------+--------------+----------+------------+----------+-----------+
```

Check if the correct class ID is set in the configuration:

> - check if types contains the correct `class_id` used in eContent for `product_groups`. It is usually 1

Reviewer: ```suggestion
- check if the `types` setting contains the correct `class_id` used in eContent for `product_groups`. It is usually 1
```

        extract: "extractText"
```

## Products are not displayed in the detail page

Make sure that the main attributes are defined:

- `name_identifier`
- `ses_sku`
- `ses_price`

## The search says products have been found but none are displayed

Check if you have issues while building the catalog elements.

> In dev environment exceptions are logged to `silver.eshop.log`.

Reviewer: ```suggestion
In dev environment, exceptions are logged to `silver.eshop.log`.
```

## How can I improve the speed of getting a product by SKU?

> Doctrine does not allow creating an index using a part of a string.

Reviewer: "The doctrine"?
```suggestion
Doctrine does not allow creating an index based on a part of a string.
```

This is why it can be helpful to setup an index manually:

``` 
create index ind_datatext  on sve_object_attributes (data_text(20));
create index ind_datatext  on sve_object_attributes_tmp (data_text(20));
```

This example assumes that the SKU has a maximum length of 20, which should be fine for almost all projects.

## Facets are lowercased

> The configuration `index_facet_fields` has to be defined for the specific fields which should not be lowercased.

Reviewer: Hard to tell if the author wanted to say: "Field names must use uppercase characters only."
